### PR TITLE
🖌️ Add new UI for mailserver selection, Add feature to disable mailserver usage

### LINKED ISSUE #11035: Allow choosing to not use a mailserver
# Problem
From a privacy perspective, it's important, for the user to be able to opt out of using a mailserver to collect history (i.e. messages sent while you were offline). 

## Implementation
Designs to be added (@errorists can you please add the final design in the comments?)

## Acceptance Criteria
User is able to disable use of a mailserver

## Notes
Giving the user control makes a stronger case in terms of privacy policy that the user controls their personal data (IP address)

## Future Steps

### PR DESCRIPTION
**Pull request by shivekkhurana**
fixes #11035 

### Summary
Connecting to a mailserver (or history node from UX perspective) logs the users IP address. This feature gives an option to opt-out. It also implements updates the UI for mailserver selection to use switch and radio buttons.

By default, mailservers usage will be turned on. Relevant changes to Status Go repo were made by @cammellos [here](https://github.com/status-im/status-go/tree/feature/use-mailservers).

### Testing notes
We need to make sure that no historical messages are fetched while joining a public chat room when `use-mailservers?` (or `use history nodes`) is disabled.

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
- Public Chats

##### Functional
- public chats
- mailservers

status: ready

### CODE REVIEW

**Comment by flexsurfer** [comment]
hey @shivekkhurana , thanks for the PR, linting is failing , you need to run `make lint` and fix all warnings , thanks

**Comment by shivekkhurana** [comment]
@flexsurfer lint warnings fixed.

**Review by cammellos — APPROVED** [review]
Looks good to me, thanks a lot for your contribution @shivekkhurana !

**Comment by cammellos** [comment]
Some e2e fails, seems like is not fetching, I will check on the PR and see what is the issue.

File: translations/en.json [hunk 1]
```diff
@@ -807,7 +809,9 @@
     "off": "Off",
     "offline": "Offline",
     "offline-messaging": "Mailserver",
-    "offline-messaging-settings": "Mailserver settings",
+    "offline-messaging-settings": "History nodes",
```
Comment (flexsurfer) [inline translations/en.json:812]: why do we change mailserver to history only here? we have to change "mailserver" to history node in all other places then? @errorists

**Comment by cammellos** [comment]
@shivekkhurana just noticed that `status-go-version.json` has not bee committed, that might explain the failed tests (`use-mailservers?` would be false by default), would you mind having a look?
Thanks

**Comment by shivekkhurana** [comment]
> @shivekkhurana just noticed that `status-go-version.json` has not bee committed, that might explain the failed tests (`use-mailservers?` would be false by default), would you mind having a look?
> Thanks

Oh, I didn't realise that it needs to be committed. Making changes right now.

**Comment by cammellos** [comment]
@shivekkhurana Another thing I noticed is that you need to set `use-mailservers?` to `true` in https://github.com/status-im/status-react/blob/44e3076022c49aee19fb8afd74ba158b131b704b/src/status_im/multiaccounts/create/core.cljs#L253 so it's available on account creation.

**Comment by shivekkhurana** [comment]
> @shivekkhurana Another thing I noticed is that you need to set `use-mailservers?` to `true` in
> 
> https://github.com/status-im/status-react/blob/44e3076022c49aee19fb8afd74ba158b131b704b/src/status_im

File: translations/en.json (hunk 1 above)
Comment (hesterbruikman) [reply translations/en.json:812]: You're right @flexsurfer I'll bundle copy issues for status-react (incl 'favourites') and will propose updates to FAQ and Privacy Policy

I think this PR meets the minimal requirement to allow users to opt-out of pinging a mailserver for now

**Comment by cammellos** [comment]
@shivekkhurana thanks for the contribution, it is scheduled to be released in 1.8.
I have created a different PR and rebased it with the latest changes here https://github.com/status-im/status-react/pull/11296 , as I wasn't able to push changes to the branch (might be a mistake on my part). I have preserved your commit.

**Comment by cammellos** [comment]
Sorry for the delay, we had a few weeks of releases and now is buidl week.

**Pull request closed without merging**